### PR TITLE
XRay Remote Sampler - Cap quotaBalance to 1x quota

### PR DESCRIPTION
**Pull request by jj22ee**
Tracking issue: #3651

Issue:
- When XRay Sampling quota is run out, [quotaBalance is refreshed](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/samplers/aws/xray/internal/reservoir.go#L72-L78).
- When quotaBalance is refreshed, it is increased by: [(seconds since last refresh) * quota](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/samplers/aws/xray/internal/reservoir.go#L106)
   - If quotaBalance is refreshed 5 seconds after previous refresh, then quotaBalance gains 5*quota. This is not desirable as it can allow for an unexpected increase burst of sampled requests after longer time periods between requests.

FIx:
Cap rule quotaBalance to 1x quota.

### CODE REVIEW

**Review by srprash — APPROVED** [review]

**Comment by jj22ee** [comment]
@Aneurysm9 Pinging for review!

**Review by Aneurysm9 — APPROVED** [review]

**Comment by jj22ee** [comment]
@Aneurysm9 
Can this be merged?

**Comment by Aneurysm9** [comment]
> @Aneurysm9 Can this be merged?

This requires at least one more approval from @open-telemetry/go-approvers before it can be merged.

**Comment by jj22ee** [comment]
@Aneurysm9 

Does this require `go-approvers` to be in the `Reviewers` list? Can this be done or how can we ping another `go-approver` to look at this PR?

**Review by MrAlias — APPROVED** [review]

**Comment by jj22ee** [comment]
@Aneurysm9 @MrAlias 
Pinging for a merge!

File: CHANGELOG.md
Comment (MrAlias) [reply CHANGELOG.md:25]: https://github.com/open-telemetry/opentelemetry-go-contrib/pull/3890

**Pull request merged**
